File: FestivalC#/Festival/Festival/Repository/Interface/IShowRepository.cs
using Festival.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Repository.Interface
{
    public interface IShowRepository : IRepository<Show, int>
    {
        /*
         * Gets all shows for a given artist 
         */
        List<Show> getShowsForArtist(Artist artist);

        /*
         * Gets all shows for a given date 
         */
        List<Show> getShowsForDate(String date);
    }
}

File: FestivalJava/Repository/src/main/java/festival/Interfaces/IShowRepository.java
package festival.Interfaces;

import festival.Domain.Show;

import java.util.List;

/**
 * Created by Sebi on 26-Mar-17.
 */
public interface IShowRepository extends IRepository<Show, Integer> {
    /*
    Gets all shows for a given artist
     */
    List<Show> getShowsForArtist(Integer idArtist);

    /*
    Gets all shows for a given date
     */
    List<Show> getShowsForDate(String date);
}
